database/migration/migrateData.ts

```diff
@@ -854,6 +856,83 @@ const migrateData = async () => {
+        CREATE OR REPLACE view permission_flattened as
+        SELECT "user".id as user_id, permission_join.organisation_id as organisation_id, permission_policy.id as permission_policy_id, template_permission.template_id as template_id  
+        FROM "user" 
+        JOIN permission_join ON permission_join.user_id = "user".id
+        JOIN permission_name ON permission_name.id = permission_join.permission_name_id
+        JOIN permission_policy ON permission_policy.id = permission_name.permission_policy_id
+        JOIN template_permission ON template_permission.permission_name_id = permission_name.id
+      `)
+
+    console.log(
+      ' - Updating policies to replace jwtPermission_array_bigint_template_ids with query rathern then in statement matching arrays in JWT'
+    )
+    const policies = (
+      await DBConnect.query({
+        text: `SELECT id, rules FROM permission_policy`,
+      })
+    ).rows as { id: number; rules: object }[]
+
+    // This is added as an optimisation, rather then doing "in" clause with array values from JWT
+    // we "select" from permission_flattened table, pre populated with userId, organisationId, permissionPolicyId and templateIds
+    // from: template_id in any (string_to_array(COALESCE(current_setting('jwt.claims.pp6_templateIds, true), '0'), ',')::integer[])
+    // to: template_id in select template_id from permission_flattened
+    //         where user_id = COALESCE(current_setting('jwt.claims.userId, true),''),'0')::integer
+    //         and permission_policy_id = 6
+    //         and (orgainisation_id = COALESCE(current_setting('jwt.claims.orgId, true),''),'0')::integer or organisation_id is null)
+    for (const row of policies) {
+      const { id, rules } = row
+
+      const jsonAsString = JSON.stringify(rules)
+      const replacement = {
+        $in: {
+          $select: {
+            $from: 'permission_flattened',
+            $where: {
+              $and: [
+                {
+                  user_id: 'jwtUserDetails_bigint_userId',
+                },
+                { permission_policy_id: id },
+                {
+                  $or: [
+                    {
+                      organisation_id: 'jwtUserDetails_bigint_orgId',
+                    },
+                    {
+                      organisation_id: {
+                        $isNull: true,
+                      },
+                    },
+                  ],
+                },
+              ],
+            },
+            template_id: true,
+          },
+        },
+      }
+      const replacementAsString = JSON.stringify(replacement)
+      const replacementAsJson = JSON.parse(
+        jsonAsString.replace(/"jwtPermission_array_bigint_template_ids"/g, replacementAsString)
+      )
+
+      await DBConnect.query({
+        text: `UPDATE permission_policy set rules = $1 where id = $2`,
+        values: [replacementAsJson, id],
+      })
+    }
+    await updateRowPolicies()
```

Comment:
This gives an error, as it's actually making a request to an endpoint that doesn't exist yet (migrate script runs before server starts). 

It's actually confusing -- there's two different `updateRowPolicies` defined -- one is in `rowLevelPolicyHelpers.ts` and is the actual function we want. But the other is in `updateRowPolicies.ts` and just queries the endpoint which runs that function. We need to refactor this so we don't keep making the same mistake (it happens on snapshot load during the docker build too) -- so you should be calling the actual `updateRowPolicies` function, and we might even be able to drop the other one (we shouldn't be calling a function that queries an endpoint in the back-end, and the front-end should just be querying the endpoint). 

I can fix this if you like, if you could just change this to import the correct version.

(In fact, we should go a step further and always run this on server restarts/snapshot reloads, so I'll change it to do that properly after this is sorted)

Reply:
Removed: 3afc7948d6291e4a0a4816b573c5ba86ce013aea